src/lib/json_copy.go

```diff
@@ -0,0 +1,29 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package lib
+
+import (
+	"encoding/json"
+)
+
+// JSONCopy copy from src to dst with json marshal and unmarshal
+func JSONCopy(dst, src interface{}) error {
```

Comment:
This can be misused when dst and src are of different types?  
If it's a short cut for your code, how about make it private to package?

Reply:
When calling this function, most of the time the `dst` and `src` are different types. If `dst` and `src` are same type, this function is not needed just assign value will copy  `src` value to `dst`.

This function is called in two different packages and many times.  2 times in `src/server/v2.0/handler/model`, 6 times in `src/server/v2.0/handler'.  Because it called in two different packages, it can't be private. And it called many times, extract it as the function is reasonable. Otherwise, I need to write many `json.Marshal` and `json.Unmarshal` statements or copy the values from one struct to another struct.

Reply:
`NOTE` comment added for this function.